From c122faa6c8c82b58dbbedaea3016fee5b1e3aae2 Mon Sep 17 00:00:00 2001
From: Jiacheng Shi <billsjc@sjtu.edu.cn>
Date: Sun, 12 Dec 2021 01:16:00 -0800
Subject: [PATCH 351/595] [Backport] vfio/iommu_type1: replace kfree with
 kvfree

mainline inclusion
from mainline-v5.16-rc6
commit 2bed2ced40c97b8540ff38df0149e8ecb2bf4c65
category: bugfix
bugzilla: 187931

Reference: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=2bed2ced40c97b8540ff38df0149e8ecb2bf4c65

--------------------------------

Variables allocated by kvzalloc should not be freed by kfree.
Because they may be allocated by vmalloc.
So we replace kfree with kvfree here.

Fixes: d6a4c185660c ("vfio iommu: Implementation of ioctl for dirty pages tracking")
Signed-off-by: Jiacheng Shi <billsjc@sjtu.edu.cn>
Link: https://lore.kernel.org/r/20211212091600.2560-1-billsjc@sjtu.edu.cn
Signed-off-by: Alex Williamson <alex.williamson@redhat.com>
Signed-off-by: Guo Xuenan <guoxuenan@huawei.com>
---
 drivers/vfio/vfio_iommu_type1.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/vfio/vfio_iommu_type1.c b/drivers/vfio/vfio_iommu_type1.c
index 1422cbb37013..b3f37a82ec0b 100644
--- a/drivers/vfio/vfio_iommu_type1.c
+++ b/drivers/vfio/vfio_iommu_type1.c
@@ -265,7 +265,7 @@ static int vfio_dma_bitmap_alloc(struct vfio_dma *dma, size_t pgsize)
 
 static void vfio_dma_bitmap_free(struct vfio_dma *dma)
 {
-	kfree(dma->bitmap);
+	kvfree(dma->bitmap);
 	dma->bitmap = NULL;
 }
 
-- 
2.17.1

